src/Mvc/Mvc.Core/src/ContentResult.cs

```diff
@@ -39,5 +44,56 @@ public override Task ExecuteResultAsync(ActionContext context)
             var executor = context.HttpContext.RequestServices.GetRequiredService<IActionResultExecutor<ContentResult>>();
             return executor.ExecuteAsync(context, this);
         }
+
+        /// <summary>
+        /// Writes the content to the HTTP response.
+        /// </summary>
+        /// <param name="httpContext">The <see cref="HttpContext"/> for the current request.</param>
+        /// <returns>A task that represents the asynchronous execute operation.</returns>
+        async Task IResult.ExecuteAsync(HttpContext httpContext)
+        {
+            if (httpContext == null)
+            {
+                throw new ArgumentNullException(nameof(httpContext));
+            }
+
+            var response = httpContext.Response;
+
+            ResponseContentTypeHelper.ResolveContentTypeAndEncoding(
+                ContentType,
+                response.ContentType,
+                DefaultContentType,
+                out var resolvedContentType,
+                out var resolvedContentTypeEncoding);
+
+            response.ContentType = resolvedContentType;
+
+            if (StatusCode != null)
+            {
+                response.StatusCode = StatusCode.Value;
+            }
+
+            var factory = httpContext.RequestServices.GetRequiredService<ILoggerFactory>();
+            var logger = factory.CreateLogger<ContentResult>();
+
+            logger.ContentResultExecuting(resolvedContentType);
+
+            if (Content != null)
+            {
+                response.ContentLength = resolvedContentTypeEncoding.GetByteCount(Content);
+                var httpResponseStreamWriterFactory = httpContext.RequestServices.GetRequiredService<IHttpResponseStreamWriterFactory>();
```

Comment:
Thanks for doing this! I cannot rely on a IHttpResponseStreamWriterFactory being registered for use with minimal Actions. Expecting the ILoggerFactory is fine. Let's use HttpResponseStreamWriter directly.

Reply:
Thanks @halter73. I've updated the code.